flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java

```diff
@@ -134,10 +138,13 @@
 	 * @return A new config options, with the given deprecated keys.
 	 */
 	public ConfigOption<T> withDeprecatedKeys(String... deprecatedKeys) {
-		FallbackKey[] fallbackKeys = Arrays.stream(deprecatedKeys)
-			.map(FallbackKey::createDeprecatedKey)
+		final Stream<FallbackKey> newFallbackKeys = Arrays.stream(deprecatedKeys).map(FallbackKey::createFallbackKey);
```

Comment:
Should be `FallbackKey::createDeprecatedKey`

Reply:
maybe rename variable to `newDeprecatedKeys`

Reply:
I'll fix the method reference.

The naming scheme is messed up generally in this context, both deprecated and fallback keys are stored as `FallbackKey` in a `fallbackKeys` field :/

I think I'll rename this one to `newDeprecatedKey`, and the union of keys to `mergedAlternativeKeys` or something...

Reply:
I'll also check if I can't add a test that deprecated keys are indeed considered deprecated.

Reply:
Good idea. Thanks Chesnay.